tests/intagration_tests.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+"""
+Authors: Fred Buchanan
+License: Apache 2.0
+Description: Intgation tests for coconut
+"""
+
+from __future__ import print_function, absolute_import, unicode_literals, division
+
+import unittest
+import sys
+import os
+import subprocess
+import shutil
+
+
+class OldCocoTest(unittest.TestCase):
+
+
+    def compile_extras(self):
+        src = os.path.join(self.src,"extras.coco")
+        subprocess.check_call(["coconut",src,self.bin])
+
+    def compile_runner(self):
+        src = os.path.join(self.src,"src","runner.coco")
+        subprocess.check_call(["coconut",src,self.bin])
+
+    def compile_agnostic(self):
+        src = os.path.join(self.src,"src","agnostic")
+        bin = os.path.join(self.bin,"cocotest")
+        subprocess.check_call(["coconut",src,bin])
+
+    def compile_2(self):
+        src = os.path.join(self.src,"src","python2")
+        bin = os.path.join(self.bin,"cocotest")
+        subprocess.check_call(["coconut","--target","2",src,bin])
+
+    def compile_3(self):
+        src = os.path.join(self.src,"src","python3")
+        bin = os.path.join(self.bin,"cocotest")
+        subprocess.check_call(["coconut","--target","3",src,bin])
+
+    def compile_35(self):
+        src = os.path.join(self.src,"src","python35")
+        bin = os.path.join(self.bin,"cocotest")
+        subprocess.check_call(["coconut","--target","35",src,bin])
+
+
+    def compile_source(self):
+        self.compile_extras()
+        self.compile_runner()
+        self.compile_agnostic()
+        
+        if sys.version_info >= (3,):
+            self.compile_3()
+        if sys.version_info >= (3,5):
+            self.compile_35()
+        if (sys.version_info >= (2,)) & (sys.version_info < (3,)):
+            self.compile_2()
+
+    def test_compile_and_run_src(self):
```

Comment:
One of the things that `coconut-travis-generator` did was go through the whole process a bunch of different times in different ways. Since I don't want to lose any coverage with this change, I'd like to replicate that here. Those ways were:

- compile each file with a separate command
- compile even the version-agnostic files with `--target`
- use `--run` and pipe `--quiet --display` to `python` as different ways of running `runner.py
- compile with `--strict`
- compile with `--minify`
- compile with `--line-numbers`
- compile with `--keep-lines`

I think the easiest way to preserve all of this behaviour is to make `integration_tests.py` accept command-line arguments, and pass those to `coconut` each time `coconut` gets called, then instead of totally gutting `coconut-travis-generator`, just replace directly compiling using `coconut` with calls to `integration_tests.py`.